UM/Qt/Bindings/MachineManagerProxy.py

```diff
@@ -166,6 +166,15 @@ def createProfile(self):
         self._manager.setActiveProfile(profile)
         return profile.getName()
 
+    @pyqtSlot(str, result = str)
+    def duplicateProfile(self, profile_name):
+        original_profile = self._manager.findProfile(profile_name, instance = self._manager.getActiveMachineInstance())
+        if not original_profile:
+            return
```

Comment:
Will this work? I think it would sometimes set the current profile to index -1 via the `selectProfile` signal in ProfilesPage.qml line 27.